File: snippets/csharp/NostrWalletConnect.cs
using Breez.Sdk.Liquid;

// ANCHOR: event-listener
public class MyNwcEventListener : NwcEventListener
{
    public void OnEvent(NwcEvent event)
    {
        switch (event.Details)
        {
            case NwcEventDetails.Connected:
                Console.WriteLine("NWC connected");
                break;
            case NwcEventDetails.Disconnected:
                Console.WriteLine("NWC disconnected");
                break;
            case NwcEventDetails.PayInvoice payInvoice:
                Console.WriteLine($"Payment {(payInvoice.Success ? "successful" : "failed")}");
                break;
            case NwcEventDetails.ListTransactions:
                Console.WriteLine("Transactions requested");
                break;
            case NwcEventDetails.GetBalance:
                Console.WriteLine("Balance requested");
                break;
        }
    }
}
// ANCHOR_END: event-listener

public class NostrWalletConnect
{
    public static void NostrWalletConnectMethod()
    {
        // ANCHOR: nwc-config
        var nwcConfig = new NwcConfig
        {
            RelayUrls = new[] { "<your-relay-url-1>" },       // Optional: Custom relay URLs (uses default if null)
            SecretKeyHex = "your-nostr-secret-key-hex"        // Optional: Custom Nostr secret key
        };
        
        var nwcService = new BindingNwcService(nwcConfig);
        
        // Add the plugin to your SDK
        var plugins = new Plugin[] { nwcService };
        // ANCHOR_END: nwc-config

        // ANCHOR: add-connection
        var connectionName = "my-app-connection";
        var connectionString = nwcService.AddConnectionString(connectionName);
        // ANCHOR_END: add-connection

        // ANCHOR: list-connections
        var connections = nwcService.ListConnectionStrings();
        // ANCHOR_END: list-connections

        // ANCHOR: remove-connection
        nwcService.RemoveConnectionString(connectionName);
        // ANCHOR_END: remove-connection

        // ANCHOR: event-management
        // Add event listener
        var listener = new MyNwcEventListener();
        var listenerId = nwcService.AddEventListener(listener);

        // Remove event listener when no longer needed
        nwcService.RemoveEventListener(listenerId);
        // ANCHOR_END: event-management

        // ANCHOR: error-handling
        try
        {
            var connectionString = nwcService.AddConnectionString("test");
            Console.WriteLine($"Connection created: {connectionString}");
        }
        catch (NwcErrorGeneric e)
        {
            Console.WriteLine($"Generic error: {e.Err}");
        }
        catch (NwcErrorPersist e)
        {
            Console.WriteLine($"Persistence error: {e.Err}");
        }
        // ANCHOR_END: error-handling
    }
}

File: snippets/python/src/nostr_wallet_connect.py
import asyncio
from breez_sdk_liquid import (NwcConfig, BindingNwcService, Plugin, NwcEventListener, NwcEvent, NwcEventDetails, NwcErrorGeneric, NwcErrorPersist
)

def nostr_wallet_connect():
    # ANCHOR: nwc-config
    nwc_config = NwcConfig(
        relay_urls=["<your-relay-url-1>"],               # Optional: Custom relay URLs (uses default if None)
        secret_key_hex="your-nostr-secret-key-hex"       # Optional: Custom Nostr secret key
    )
    
    nwc_service = BindingNwcService(nwc_config)
    
    # Add the plugin to your SDK
    plugins = [nwc_service]
    # ANCHOR_END: nwc-config

    # ANCHOR: add-connection
    connection_name = "my-app-connection"
    connection_string = nwc_service.add_connection_string(connection_name)
    # ANCHOR_END: add-connection

    # ANCHOR: list-connections
    connections = nwc_service.list_connection_strings()
    # ANCHOR_END: list-connections

    # ANCHOR: remove-connection
    nwc_service.remove_connection_string(connection_name)
    # ANCHOR_END: remove-connection

    # ANCHOR: event-listener
    class MyNwcEventListener(NwcEventListener):
        def on_event(self, event: NwcEvent):
            if isinstance(event.details, NwcEventDetails.Connected):
                print("NWC connected")
            elif isinstance(event.details, NwcEventDetails.Disconnected):
                print("NWC disconnected")
            elif isinstance(event.details, NwcEventDetails.PayInvoice):
                success = event.details.success
                print(f"Payment {'successful' if success else 'failed'}")
            elif isinstance(event.details, NwcEventDetails.ListTransactions):
                print("Transactions requested")
            elif isinstance(event.details, NwcEventDetails.GetBalance):
                print("Balance requested")
    # ANCHOR_END: event-listener

    # ANCHOR: event-management
    # Add event listener
    listener = MyNwcEventListener()
    listener_id = nwc_service.add_event_listener(listener)

    # Remove event listener when no longer needed
    nwc_service.remove_event_listener(listener_id)
    # ANCHOR_END: event-management

    # ANCHOR: error-handling
    try:
        connection_string = nwc_service.add_connection_string("test")
        print(f"Connection created: {connection_string}")
    except NwcErrorGeneric as e:
        print(f"Generic error: {e.err}")
    except NwcErrorPersist as e:
        print(f"Persistence error: {e.err}")
    except Exception as e:
        print(f"Unknown error: {e}")
    # ANCHOR_END: error-handling